File: pyg_user_web/src/main/webapp/js/controller/userController.js
app.controller("userController",function ($scope,userService) {
  
    $scope.entity = {sourceType:'1'}


    $scope.sendSms = function () {
        userService.sendSms($scope.entity.phone).success(function (response) {
            if (response.success) {
                alert("倒计时效果");
            } else {
                alert(response.message);
            }
        })
    }
    
    $scope.register = function () {
        userService.register($scope.entity,$scope.code).success(function (response) {
            if (response.success) {
                alert("跳转到登陆页面");
            } else {
                alert(response.message);
            }
        })
    }
})

File: pyg_user_web/src/main/java/com/pyg/user/controller/UserController.java
package com.pyg.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pyg.pojo.TbUser;
import com.pyg.user.service.UserService;
import entity.Result;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/user")
public class UserController {

    @Reference
    private UserService userService;

    @RequestMapping("/sendSms")
    public Result sendSms(String phone) {
        //
        try {
            userService.sendSms(phone);
            return new Result(true,"发送成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"发送失败");
        }
    }

    @RequestMapping("/register")
    public Result register(@RequestBody TbUser user, String code) {
        try {
            userService.register(user,code);
            return new Result(true,"");
        }catch (RuntimeException e) {
            return new Result(false,e.getMessage());
        }
        catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"注册失败");
        }
    }
}
